src/main/java/com/icthh/xm/ms/entity/web/rest/error/EntityExceptionTranslator.java

```diff
@@ -0,0 +1,57 @@
+package com.icthh.xm.ms.entity.web.rest.error;
+
+import com.google.common.base.Throwables;
+import com.icthh.xm.commons.i18n.error.domain.vm.ErrorVM;
+import com.icthh.xm.commons.i18n.error.domain.vm.ParameterizedErrorVM;
+import com.icthh.xm.commons.i18n.error.web.ExceptionTranslator;
+import com.icthh.xm.commons.i18n.spring.service.LocalizationMessageService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.sql.SQLException;
+
+@Slf4j
+@ControllerAdvice
+@RequiredArgsConstructor
+@Order(value = Ordered.HIGHEST_PRECEDENCE)
+public class EntityExceptionTranslator {
+
+    private static final String INTEGRITY_CONSTRAINT_VIOLATION_GROUP_CODE = "23";
+    private static final String INTEGRITY_CONSTRAINT_VIOLATION_MESSAGE_CODE = "error.db.";
```

Comment:
as we are handling only DataIntegrityViolationException suggest to make error code prefix as: 
`error.db.dataIntegrityViolation`